packages/amplify-migration/src/app_functions_definition_fetcher.ts

```diff
@@ -0,0 +1,82 @@
+import assert from 'node:assert';
+import { FunctionDefinition } from '@aws-amplify/amplify-gen2-codegen';
+import { getFunctionDefinition } from '@aws-amplify/amplify-gen1-codegen-function-adapter';
+import { BackendEnvironmentResolver } from './backend_environment_selector';
+import { LambdaClient, GetFunctionCommand } from '@aws-sdk/client-lambda';
+import { StateManager } from '@aws-amplify/amplify-cli-core';
+
+export interface AppFunctionsDefinitionFetcher {
+  getDefinition(): Promise<FunctionDefinition[] | undefined>;
+}
+
+export class AppFunctionsDefinitionFetcher {
+  constructor(
+    private lambdaClient: LambdaClient,
+    private backendEnvironmentResolver: BackendEnvironmentResolver,
+    private stateManager: StateManager,
+  ) {}
+
+  getDefinition = async (): Promise<FunctionDefinition[] | undefined> => {
+    const backendEnvironment = await this.backendEnvironmentResolver.selectBackendEnvironment();
+    assert(backendEnvironment?.stackName);
+
+    const meta = this.stateManager.getMeta();
+
```

Comment:
can we add an assert for `meta` to make sure is not nullable/undefined? We can then remove the `?` in the below statements.

Reply:
added